src/Stratis.SmartContracts.Core/Extensions.cs

```diff
@@ -14,10 +15,15 @@ public static string ToHexString(this byte[] ba)
 
         public static byte[] HexToByteArray(this string hex)
         {
-            int NumberChars = hex.Length;
+            string toHex = hex;
+
+            if (hex.StartsWith("0x"))
+                toHex = hex.Substring(2);
+
+            int NumberChars = toHex.Length;
             byte[] bytes = new byte[NumberChars / 2];
```

Comment:
This will not behave as expected if NumberChars is not even. We should consider zero-padding the hex if that's the case.

Reply:
Ah, I see there's a unit test for that already. I think zero-padding is probably preferable over outright failure.

Reply:
Fair point.